src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -249,6 +249,7 @@ def add_noise(
         noise: Union[torch.FloatTensor, np.ndarray],
         timesteps: Union[torch.IntTensor, np.ndarray],
     ) -> Union[torch.FloatTensor, np.ndarray]:
+        timesteps = timesteps.to(self.alphas_cumprod.device)
```

Comment:
This breaks the assumption that we can work with numpy arrays as well :( 
Maybe we could add a device argument to `scheduler.set_format()` and call if from `pipeline.to()`?

Reply:
I'd rather minimize the changes across files just for this particular case that only affects `mps`. How about some defensive coding like:
- Check the format is `pt`
- And `alphas_cumprod` is actually a tensor

Then move using `to()` ?

Reply:
Think we'll move away from `set_format()` since we'll soon have framework dependent schedulers - so I think this is ok here